src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -124,81 +124,118 @@ export default class SourceFileLinter {
-						if (ts.isClassDeclaration(declaration)) {
-							if (declaration.name?.getText() === "ManagedObject" &&
+						if (!ts.isClassDeclaration(declaration)) {
+							return false;
+						}
+						for (const baseClass of baseClasses) {
+							if (declaration.name?.getText() === baseClass.name &&
 								(
-									// Declaration via type definitions
+								// Declaration via type definitions
 									(
 										declaration.parent.parent &&
 										ts.isModuleDeclaration(declaration.parent.parent) &&
-										declaration.parent.parent.name?.text === "sap/ui/base/ManagedObject"
+										declaration.parent.parent.name?.text === baseClass.module
 									) ||
 									// Declaration via real class (within sap.ui.core project)
 									(
 										ts.isSourceFile(declaration.parent) &&
-										declaration.parent.fileName === "/resources/sap/ui/base/ManagedObject.js"
+										declaration.parent.fileName === `/resources/${baseClass.module}.js`
 									)
 								)
 							) {
 								return true;
-							} else {
-								return isObjectMetadataAncestor(declaration);
 							}
-						} else {
-							return false;
 						}
+						return isClassUi5Subclass(declaration);
 					});
 				});
 			// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
 			}).reduce((acc, cur) => cur || acc, false) ?? false;
 		};
 
-		const metadataFound = taskStart("isPropertyInMetadata", this.#resourcePath, true);
+		return isClassUi5Subclass(node);
+	}
 
-		if (node.name.getText() !== "metadata") {
-			metadataFound();
-			return false;
-		}
+	analyzeControlRendererDeclaration(node: ts.ClassDeclaration) {
+		const className = node.name?.getText() ?? "<unknown>";
+		const rendererMember = node.members.find((member) => {
+			return (ts.isPropertyDeclaration(member) || ts.isMethodDeclaration(member)) &&
+				member.modifiers?.some((modifier) => modifier.kind === ts.SyntaxKind.StaticKeyword) &&
+				member.name.getText() === "renderer";
+		});
 
-		let parentNode: ts.Node = node.parent;
-		while (parentNode && parentNode.kind !== ts.SyntaxKind.ClassDeclaration) {
-			parentNode = parentNode.parent;
+		if (!rendererMember) {
+			// Special cases: Some base classes do not require sub-classes to have a renderer defined:
+			if (this.isUi5ClassDeclaration(node, [
+				"sap/ui/core/mvc/View",
+				// XMLComposite is deprecated, but there still shouldn't be a false-positive about a missing renderer
+				"sap/ui/core/XMLComposite",
+				"sap/ui/core/webc/WebComponent",
+				"sap/uxap/BlockBase",
+			])) {
+				return;
+			}
+			// No definition of renderer causes the runtime to load the corresponding Renderer module synchronously
+			this.#reporter.addMessage(MESSAGE.MISSING_CONTROL_RENDERER_DECLARATION, {className}, node);
+			return;
 		}
 
-		if (!parentNode) {
-			metadataFound();
-			return false;
-		}
+		if (ts.isPropertyDeclaration(rendererMember) && rendererMember.initializer) {
```

Comment:
Somehow my last comment went lost, strange.

I wondered why the filter above accepts MethodDeclarations but the check here does not care about them.
Is that because the check for apiVersion:2 is only part of the follow-up BLI?

Reply:
First I had the check but then removed the handling of MethodDeclarations as they are not considered deprecated. I missed to adjust the filter, as it actually wasn't on prupose.

But you're right. When we check for the renderer itself, it probably needs to be added in there again, so I would just leave it there.